presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -0,0 +1,1970 @@
+            if (!partition.isPresent()) {
+                throw new PartitionNotFoundException(schemaTableName, partitionValues);
+            }
+            partitionActionsOfTable.put(
+                    partitionValues,
+                    new Action<>(ActionType.INSERT_EXISTING, new PartitionAndMore(partition.get(), currentLocation, Optional.of(fileNames)), session.getUser(), session.getQueryId()));
+            return;
+        }
+
+        switch (oldPartitionAction.getType()) {
+            case DROP:
+                throw new PartitionNotFoundException(schemaTableName, partitionValues);
+            case ADD:
+            case ALTER:
+            case INSERT_EXISTING:
+                throw new UnsupportedOperationException("Inserting into a partition that were added, altered, or inserted into in the same transaction is not supported");
+            default:
+                throw new IllegalStateException("Unknown action type");
+        }
+    }
+
+    public synchronized Set<String> getRoles(String user)
+    {
+        checkReadable();
+        return delegate.getRoles(user);
+    }
+
+    public synchronized Set<HivePrivilegeInfo> getDatabasePrivileges(String user, String databaseName)
+    {
+        checkReadable();
+        return delegate.getDatabasePrivileges(user, databaseName);
+    }
+
+    public synchronized Set<HivePrivilegeInfo> getTablePrivileges(String user, String databaseName, String tableName)
+    {
+        checkReadable();
+        return delegate.getTablePrivileges(user, databaseName, tableName);
+    }
+
+    public synchronized void grantTablePrivileges(String databaseName, String tableName, String grantee, Set<PrivilegeGrantInfo> privilegeGrantInfoSet)
+    {
+        setExclusive((delegate, hdfsEnvironment) -> delegate.grantTablePrivileges(databaseName, tableName, grantee, privilegeGrantInfoSet));
+    }
+
+    public synchronized void revokeTablePrivileges(String databaseName, String tableName, String grantee, Set<PrivilegeGrantInfo> privilegeGrantInfoSet)
+    {
+        setExclusive((delegate, hdfsEnvironment) -> delegate.revokeTablePrivileges(databaseName, tableName, grantee, privilegeGrantInfoSet));
+    }
+
+    public synchronized void declareIntentionToWrite(ConnectorSession session, WriteMode writeMode, Path stagingPathRoot, String filePrefix, SchemaTableName schemaTableName)
+    {
+        setShared();
+        if (writeMode == WriteMode.DIRECT_TO_TARGET_EXISTING_DIRECTORY) {
+            Map<List<String>, Action<PartitionAndMore>> partitionActionsOfTable = partitionActions.get(schemaTableName);
+            if (partitionActionsOfTable != null && !partitionActionsOfTable.isEmpty()) {
+                throw new PrestoException(NOT_SUPPORTED, "Inserting into a table whose partition was modified in the same transaction is not supported when Presto is configured to skip temporary directories.");
```

Comment:
Avoid the word `whose` as it personifies the actor.  Instead try `Can not insert into a table with a partition that has been modified..`